HHS/Head-Start-TTADP
# [TTAHUB-3113] merge duped AROs

## Description of change

This merges all the duplicate AROs we've made in history using almost the same logic as for a previous support request in https://jira.acf.gov/browse/TTAHUB-3309

It also drop the current non-unique index and replaces it with a _unique_ index for `goalId` and `activityReportId`

## How to test

There's not an easy way to see in on the frontend, but the results should look similar to 
```
            table_name            | updated_count | inserted_count | deleted_count
----------------------------------+---------------+----------------+---------------
 ActivityReportObjectiveTopics    |          1412 |             73 |          2189
 ActivityReportObjectiveCourses   |             8 |              0 |             2
 ActivityReportObjectiveFiles     |          1381 |            890 |          5318
 ActivityReportObjectiveResources |          1467 |            192 |          2827
 ActivityReportObjectives         |           596 |              0 |          2517
 ```
 Also you can run `SELECT arid, oid FROM (SELECT "activityReportId" arid, "objectiveId" oid, COUNT(*) cnt FROM "ActivityReportObjectives" GROUP BY 1,2) a WHERE cnt > 1;` to confirm there are no duplicates left

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3113

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
